File: examples/src/VBA/ExtractingVBAMacros.cs
import aspose.slides as slides
using Aspose.slides.Examples.CSharp
using Aspose.slides.Vba
using System
using System.Collections.Generic
using System.Linq
using System.text
using System.Threading.Tasks

namespace CSharp.VBA
{
    class ExtractingVBAMacros
    {
        public static void Run()
        {
            #ExStart:ExtractingVBAMacros

            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_VBA()

            using (Presentation pres = new Presentation(dataDir + "VBA.pptm"))
            {
                if (pres.VbaProject != None) # check if Presentation contains VBA Project
                {
                    foreach (IVbaModule module in pres.VbaProject.Modules)
                    {
                        print(module.Name)
                        print(module.SourceCode)
                    }
                }
            }

            #ExEnd:ExtractingVBAMacros

        }
    }
}

File: examples/src/VBA/extracting_vba_macros.py
import aspose.slides as slides


def extracting_vba_macros(global_opts):
    with slides.Presentation(global_opts.data_dir + "VBA.pptm") as pres:
        # check if Presentation contains VBA Project
        if pres.vba_project is not None:
            for module in pres.vba_project.modules:
                print(module.name)
                print(module.source_code)
